### ISSUE TITLE
Option for replacing memory allocation and file I/O

### ISSUE THREAD

**Issue by rationalcoder** [syoyo/tinyobjloader-c#8]
The STB style is good, but it's incomplete. There should be an option to provide file data without `<stdio.h>`and memory allocation without `malloc()`/`realloc()`/`free()`. See the STB libraries for examples.

**Comment by syoyo** [comment]
Yeah, its possible but require some amount of work(I want  also remove other libc functions like `sscanf`)

PR is always welcome!
